src/main/java/seedu/financialplanner/utils/Ui.java

```diff
@@ -134,4 +135,40 @@ public void printResetBudget() {
     public void printDisplayChart(String type, String chart) {
         showMessage("Displaying " + chart + "chart for " + type);
     }
+
+    public void printOverview(String... args) {
+        String balance = args[0];
+        String income = args[1];
+        String expense = args[2];
+        String budget = args[3];
+        String reminders = args[4];
+
+        showMessage("Here is an overview of your financials:\n" +  "Total balance: " + balance + "\n" +
+                "Highest income: " + income + "\n" + "Highest expense: " + expense + "\n" +
+                "Remaining budget for the month: " + budget + "\n\n" + "Reminders:\n" + reminders);
+    }
+
+    public void printSetBudget() {
+        showMessage("A monthly budget of " + Budget.getInitialBudgetString() + " has been set.");
+    }
+
+    public void printBudgetExceedBalance() {
+        showMessage("Since initial budget exceeds current balance, budget will be reset to current balance.");
+    }
+
+    public void printBudgetError(String errorType) {
+        switch (errorType) {
+        case "delete":
+            showMessage("Budget has not been set yet.");
+            break;
+        case "reset":
+            showMessage("Budget has not been spent yet.");
+            break;
+        case "view":
+            showMessage("There is no existing budget.");
+            break;
+        default:
+            showMessage("Unknown command");
```

Comment:
can logging be added here if default case is not meant to be reached

Reply:
thanks for reminding i missed that

Reply:
added